pinot-core/src/main/java/org/apache/pinot/core/query/reduce/ExecutionStatsAggregator.java

```diff
@@ -275,83 +143,119 @@ public void setStats(@Nullable String rawTableName, BrokerResponseNative brokerR
+      brokerResponseNative.setRealtimeThreadCpuTimeNs(getLongValue(DataTable.MetadataKey.THREAD_CPU_TIME_NS));
+      brokerResponseNative.setRealtimeSystemActivitiesCpuTimeNs(
+          getLongValue(DataTable.MetadataKey.SYSTEM_ACTIVITIES_CPU_TIME_NS));
+      brokerResponseNative.setRealtimeResponseSerializationCpuTimeNs(
+          getLongValue(DataTable.MetadataKey.RESPONSE_SER_CPU_TIME_NS));
+      brokerResponseNative.setRealtimeTotalCpuTimeNs(brokerResponseNative.getRealtimeThreadCpuTimeNs()
+          + brokerResponseNative.getRealtimeSystemActivitiesCpuTimeNs()
+          + brokerResponseNative.getRealtimeResponseSerializationCpuTimeNs());
     }
-    brokerResponseNative.setNumConsumingSegmentsProcessed(_numConsumingSegmentsProcessed);
-    brokerResponseNative.setNumConsumingSegmentsMatched(_numConsumingSegmentsMatched);
 
     // Update broker metrics.
     if (brokerMetrics != null && rawTableName != null) {
-      brokerMetrics.addMeteredTableValue(rawTableName, BrokerMeter.DOCUMENTS_SCANNED, _numDocsScanned);
-      brokerMetrics.addMeteredTableValue(rawTableName, BrokerMeter.ENTRIES_SCANNED_IN_FILTER,
-          _numEntriesScannedInFilter);
-      brokerMetrics.addMeteredTableValue(rawTableName, BrokerMeter.ENTRIES_SCANNED_POST_FILTER,
-          _numEntriesScannedPostFilter);
-      brokerMetrics.addTimedTableValue(rawTableName, BrokerTimer.OFFLINE_THREAD_CPU_TIME_NS, _offlineThreadCpuTimeNs,
-          TimeUnit.NANOSECONDS);
-      brokerMetrics.addTimedTableValue(rawTableName, BrokerTimer.REALTIME_THREAD_CPU_TIME_NS, _realtimeThreadCpuTimeNs,
-          TimeUnit.NANOSECONDS);
-      brokerMetrics.addTimedTableValue(rawTableName, BrokerTimer.OFFLINE_SYSTEM_ACTIVITIES_CPU_TIME_NS,
-          _offlineSystemActivitiesCpuTimeNs, TimeUnit.NANOSECONDS);
-      brokerMetrics.addTimedTableValue(rawTableName, BrokerTimer.REALTIME_SYSTEM_ACTIVITIES_CPU_TIME_NS,
-          _realtimeSystemActivitiesCpuTimeNs, TimeUnit.NANOSECONDS);
-      brokerMetrics.addTimedTableValue(rawTableName, BrokerTimer.OFFLINE_RESPONSE_SER_CPU_TIME_NS,
-          _offlineResponseSerializationCpuTimeNs, TimeUnit.NANOSECONDS);
-      brokerMetrics.addTimedTableValue(rawTableName, BrokerTimer.REALTIME_RESPONSE_SER_CPU_TIME_NS,
-          _realtimeResponseSerializationCpuTimeNs, TimeUnit.NANOSECONDS);
-      brokerMetrics.addTimedTableValue(rawTableName, BrokerTimer.OFFLINE_TOTAL_CPU_TIME_NS, _offlineTotalCpuTimeNs,
-          TimeUnit.NANOSECONDS);
-      brokerMetrics.addTimedTableValue(rawTableName, BrokerTimer.REALTIME_TOTAL_CPU_TIME_NS, _realtimeTotalCpuTimeNs,
-          TimeUnit.NANOSECONDS);
-
-      if (_minConsumingFreshnessTimeMs != Long.MAX_VALUE) {
-        brokerMetrics.addTimedTableValue(rawTableName, BrokerTimer.FRESHNESS_LAG_MS,
-            System.currentTimeMillis() - _minConsumingFreshnessTimeMs, TimeUnit.MILLISECONDS);
-      }
+      addBrokerMetrics(rawTableName, brokerMetrics, brokerResponseNative);
```

Comment:
is one of the 2 for backward compatibility purpose? it seems like they add the same metrics keys.

let's mark one of them as TODO clean up after next release.

Reply:
No `addBrokerMetrics` is for reporting grafana metrics not Broker response.

Reply:
ah. nice. thanks for the explanation!